front/src/modules/trainschedule/components/ImportTrainSchedule/ImportTrainScheduleConfig.tsx

```diff
@@ -41,6 +47,69 @@ const ImportTrainScheduleConfig = ({
   const dispatch = useAppDispatch();
   const { openModal, closeModal } = useContext(ModalContext);
 
+  const [postSearch] = osrdEditoastApi.endpoints.postSearch.useMutation();
+
+  // Function to perform the search using ciCode and chCode
+  const searchOperationalPoints = async (
+    ciCode: string,
+    chCode?: string
+  ): Promise<SearchResultItemOperationalPoint[]> => {
+    const isSearchingByTrigram = !Number.isInteger(+ciCode) && ciCode.length < 4;
+    const searchQuery = isSearchingByTrigram
+      ? ['or', ['=i', ['trigram'], ciCode], ['search', ['name'], ciCode]]
+      : ['or', ['search', ['name'], ciCode], ['like', ['to_string', ['uic']], `%${ciCode}%`]];
+
+    const payload = {
+      object: 'operationalpoint',
+      query: ['and', searchQuery],
+    };
+
+    try {
+      const results = await postSearch({ searchPayload: payload, pageSize: 101 }).unwrap();
+
+      const filteredResults = (results as SearchResultItemOperationalPoint[]).filter(
+        (result) => !chCode || result.ch.toLowerCase().includes(chCode.toLowerCase())
+      );
+
+      return filteredResults;
+    } catch (error: any) {
+      console.error('Search failed:', error);
+
+      if (error.status === 'PARSING_ERROR' && error.originalStatus === 502) {
+        console.error('Received 502 Bad Gateway error:', error.data || error.error);
+      } else {
+        console.error('Unexpected error occurred:', error);
+      }
+
+      return [];
+    }
+  };
+
+  const overloadDict = async (localCichDict: Record<string, any>): Promise<Record<string, any>> => {
+    const newDict = { ...localCichDict };
+
+    for (const key of Object.keys(newDict)) {
+      const { ciCode, chCode } = newDict[key];
+
+      // Call the search function with ciCode and chCode
+      const searchResult = await searchOperationalPoints(ciCode, chCode);
```

Comment:
This sends one OP search HTTP request per cich. Would be nicer to send a single HTTP request for all cichs. See the OSRD → NGE converter for an example.